### ISSUE TITLE
[Issue]: MI250X SGEMM performance

### ISSUE THREAD

**Issue by cctry** [ROCm/rocWMMA#362]
### Problem Description

Hi,

I am testing with sample/perf_sgemm.cpp on MI250X with ROCM/5.7.1. 
When the problem size is (4096, 4096, 128), the achieved throughput is only ~13 TFLOPS, which is below 30% of the peak.
Is this performance expected for this problem?

### Operating System

SUSE

### CPU

AMD EPYC 7A53

### GPU

AMD Instinct MI250X

### ROCm Version

ROCm 5.7.1

**Comment by cgmillette** [comment]
Hello @cctry! Thanks for your question.
TLDR answer is yes, a lower performance would be expected for this problem.

One thing we can keep in mind is that the typical GEMM kernel marches through the problem's entire K dimension. In this case K is very short which can be a bit problematic:
- The overhead cost of launching wavefronts of the grid is higher relative to the kernel length itself (now extremely short). This weighs more in shorter kernels and will take away from the 'Speed of Light' calculation. The first wavefront will also probably finish long before the last wavefront which extends the timing range.
- The wavefronts don't have time to reach what we can call 'steady state'. Steady state can be thought of as regularized patterns of resource usage by wavefronts, which can be pipelined efficiently if their timing is regular and balanced. Otherwise there will be regular thrashing and higher resource contention.
- Sometimes caching could become less efficient as well, due to the lower likelihood of neighboring wavefronts accessing the same area of memory at the same time.

**Comment by cctry** [comment]
Thank you
